blockchain/storagev2/mdbx/mdbx.go

```diff
@@ -0,0 +1,180 @@
+package mdbx
+
+import (
+	"os"
+
+	"github.com/0xPolygon/polygon-edge/blockchain/storagev2"
+	"github.com/erigontech/mdbx-go/mdbx"
+	"github.com/hashicorp/go-hclog"
+)
+
+type MdbxOpts struct {
+	path  string
+	flags uint
+}
+
+// MdbxDB is the mdbx implementation of the kv storage
+type MdbxDB struct {
+	env *mdbx.Env
+	dbi [storagev2.MAX_TABLES]mdbx.DBI
+}
+
+var tableMapper = map[uint8]string{
+	storagev2.BODY:         "Body",
+	storagev2.CANONICAL:    "Canonical",
+	storagev2.DIFFICULTY:   "Difficulty",
+	storagev2.HEADER:       "Header",
+	storagev2.RECEIPTS:     "Receipts",
+	storagev2.FORK:         "Fork",
+	storagev2.HEAD_HASH:    "HeadHash",
+	storagev2.HEAD_NUMBER:  "HeadNumber",
+	storagev2.BLOCK_LOOKUP: "BlockLookup",
+	storagev2.TX_LOOKUP:    "TxLookup",
+}
+
+// NewMdbxStorage creates the new storage reference for mdbx database
+func NewMdbxStorage(path string, logger hclog.Logger) (*storagev2.Storage, error) {
+	var dbs [2]storagev2.Database
+
+	// Set default options
+	opts := &MdbxOpts{
+		path: path,
+	}
+
+	env, err := mdbx.NewEnv()
+	if err != nil {
+		return nil, err
+	}
+
+	if err = env.SetOption(mdbx.OptMaxDB, uint64(storagev2.MAX_TABLES)); err != nil {
+		return nil, err
+	}
+
+	if err = env.SetGeometry(-1, -1, 2*1024*1024*1024*1024, 2*1024*1024*1024, -1, int(defaultPageSize())); err != nil {
+		return nil, err
+	}
+
+	err = env.Open(opts.path, opts.flags, 0664)
+	if err != nil {
+		return nil, err
+	}
+
+	db := &MdbxDB{
+		env: env,
+	}
+
+	if err := db.openDBI(0); err != nil {
+		return nil, err
+	}
+
+	dbs[0] = db
+	dbs[1] = nil
+
+	return storagev2.Open(logger.Named("mdbx"), dbs)
+}
+
+func defaultPageSize() uint64 {
+	osPageSize := os.Getpagesize()
+	if osPageSize < 4096 { // reduce further may lead to errors (because some data is just big)
+		osPageSize = 4096
+	} else if osPageSize > mdbx.MaxPageSize {
+		osPageSize = mdbx.MaxPageSize
+	}
+
+	osPageSize = osPageSize / 4096 * 4096 // ensure it's rounded
+
+	return uint64(osPageSize)
+}
+
+func (db *MdbxDB) view(f func(tx *mdbx.Txn) error) (err error) {
+	// can't use db.env.View method - because it calls commit for read transactions - it conflicts with write transactions.
+	tx, err := db.env.BeginTxn(nil, mdbx.Readonly)
+	if err != nil {
+		return err
+	}
+
+	return f(tx)
+}
+
+func (db *MdbxDB) update(f func(tx *mdbx.Txn) error) (err error) {
+	tx, err := db.env.BeginTxn(nil, 0)
+	if err != nil {
+		return err
+	}
+
+	err = f(tx)
+	if err != nil {
+		return err
+	}
+
+	_, err = tx.Commit()
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
Just return an err variable after `_, err = tx.Commit()`

```suggestion
	return err
```